src/game_player.cpp

```diff
@@ -202,6 +200,16 @@ void Game_Player::UpdateSelfMovement() {
 		if (IsMovable()) {
 			const auto old_x = GetX();
 			const auto old_y = GetY();
+			const bool force_through = (Player::debug_flag
+					&& Input::IsPressed(Input::DEBUG_THROUGH)
+					&& !GetThrough());
+			if (force_through) {
+				SetThrough(true);
+			}
+			auto sg = makeScopeGuard([&](){
```

Comment:
is the usage of a scope guard really needed here?
There is only one exit path and we don't throw/catch exceptions.
Though what actually bugs me is that this comes from liblcf. We already have other borderline-stuff in liblcf (INI) we call from Player but imo here the header should be copied into Player and made a private/not-installed header in liblcf.

Reply:
I always get paranoid whenever there is a thing like this where you temporarily set some state, call a function, and then restore it. This gets optimized away and we ever change the code later to throw it's still safe.

Overly paranoid for sure but that's usually how I do these things as a habit. Can take it away if its really objectionable.

Reply:
I understand the intention but I want to discourage that liblcf becomes a general collection of convenience headers that are used by Player because we don't use C++17.
So I would be already happy if you put a copy of scopeguard into the Player src folder, too.